src/containers/AccountModal/forms/ChooseOffer.tsx

```diff
@@ -1,109 +1,71 @@
-import React, { useEffect, useMemo } from 'react';
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
 import { mixed, object, SchemaOf } from 'yup';
 import { useTranslation } from 'react-i18next';
-import { useQueries, useQuery } from 'react-query';
 import { useHistory } from 'react-router';
 import shallow from 'zustand/shallow';
 
+import useOffers from '../../../hooks/useOffers';
+
 import { addQueryParam, removeQueryParam } from '#src/utils/history';
-import { useConfigStore } from '#src/stores/ConfigStore';
 import { useCheckoutStore } from '#src/stores/CheckoutStore';
-import { getOffer } from '#src/services/checkout.service';
 import LoadingOverlay from '#src/components/LoadingOverlay/LoadingOverlay';
 import ChooseOfferForm from '#src/components/ChooseOfferForm/ChooseOfferForm';
 import useForm, { UseFormOnSubmitHandler } from '#src/hooks/useForm';
-import type { ChooseOfferFormData, OfferPeriodicity, OfferType } from '#types/account';
-import type { Offer } from '#types/checkout';
+import type { ChooseOfferFormData, OfferType } from '#types/account';
 
 const ChooseOffer = () => {
   const history = useHistory();
   const { t } = useTranslation('account');
-  const config = useConfigStore(({ config }) => config);
-  const { cleengSandbox, json } = config;
-  const { setOffer, setOfferType } = useCheckoutStore(({ setOffer, setOfferType }) => ({ setOffer, setOfferType }), shallow);
-
-  const cleengMonthlyOffer = json?.cleengMonthlyOffer as string;
-  const cleengYearlyOffer = json?.cleengYearlyOffer as string;
-
-  const { requestedMediaOffers } = useCheckoutStore(({ requestedMediaOffers }) => ({ requestedMediaOffers: requestedMediaOffers || [] }), shallow);
-
-  // `useQueries` is not strongly typed :-(
-  const { data: monthlyOfferData, isLoading: isMonthlyOfferLoading } = useQuery(['offer', cleengMonthlyOffer], () =>
-    getOffer({ offerId: cleengMonthlyOffer }, cleengSandbox),
-  );
-  const { data: yearlyOfferData, isLoading: isYearlyOfferLoading } = useQuery(['offer', cleengYearlyOffer], () =>
-    getOffer({ offerId: cleengYearlyOffer }, cleengSandbox),
-  );
-
-  const tvodOfferQueries = useQueries(
-    requestedMediaOffers?.map(({ offerId }) => ({
-      queryKey: ['offer', offerId],
-      queryFn: () => getOffer({ offerId }, cleengSandbox),
-      enabled: !!offerId,
-    })),
-  );
+  const { setOffer } = useCheckoutStore(({ setOffer }) => ({ setOffer }), shallow);
+  const { tvodOffers, yearlyOffer, monthlyOffer, hasPremierOffer, isLoadingOffers } = useOffers();
+  const allOffers = useMemo(() => [yearlyOffer, monthlyOffer, ...tvodOffers], [yearlyOffer, monthlyOffer, tvodOffers]);
```

Comment:
Kind of weird how we split the offers in the hook and then immediately recombine them here

Reply:
This is refactored as well. This one did throw me back a bit since all logic and components relied on these "separate offers".